pkg/controller/constraint/constraint_controller.go

```diff
@@ -153,12 +203,58 @@ func (r *ReconcileConstraint) Reconcile(request reconcile.Request) (reconcile.Re
 			if err := r.Update(context.Background(), instance); err != nil {
 				return reconcile.Result{Requeue: true}, nil
 			}
+			// removing constraint entry from cache
+			r.deleteConstraintKey(constraintKey)
 		}
 	}
-
+	reportMetrics = true
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileConstraint) addConstraintKey(constraintKey string, t Tags) {
+	r.constraintsCacheMux.Lock()
+	defer r.constraintsCacheMux.Unlock()
+
+	r.constraintsCache[constraintKey] = Tags{
+		enforcementAction: t.enforcementAction,
+		status:            t.status,
+	}
+}
+
+func (r *ReconcileConstraint) deleteConstraintKey(constraintKey string) {
+	r.constraintsCacheMux.Lock()
+	defer r.constraintsCacheMux.Unlock()
+
+	delete(r.constraintsCache, constraintKey)
+}
+
+func (r *ReconcileConstraint) reportTotalConstraints() {
+	r.constraintsCacheMux.Lock()
```

Comment:
This should be a read lock to avoid unnecessarily blocking multiple calls to reportTotalConstraints()

Reply:
updated, thanks